src/Engine/ProtoCore/RuntimeData.cs

```diff
@@ -145,41 +159,17 @@ public CallSite GetCallSite(int classScope, string methodName, Executable execut
             if (!nodeGuids.Any()) // Nothing to persist now.
                 return nodeMap;
 
-            // Attempt to get the list of graph node if one exists.
-            IEnumerable<GraphNode> graphNodes = null;
+            List<CallSite> callsites;
+            foreach (Guid nodeGuid in nodeGuids)
             {
-                if (executable != null)
+                if (NodeToCallsiteObjectMap.TryGetValue(nodeGuid, out callsites))
                 {
-                    var stream = executable.instrStreamList;
-                    if (stream != null && (stream.Length > 0))
-                    {
-                        var graph = stream[0].dependencyGraph;
-                        if (graph != null)
-                            graphNodes = graph.GraphList;
```

Comment:
@aparajit-pratap can you give us a summary of what a `GraphNode` is in the context of the VM - and then is there any reason not to rename this to something less confusing as soon as possible?

Reply:
1. During codegen phase the compiler generates `graphnode`s that represent each executable statement
2. `Graphnode`s contain information about dependencies between statements​ and runtime properties instructing VM which instruction to execute next​
3. They have a property that points to start and end of instruction block​
4. Every time a graph node executes it jumps to its target label as per dependency property stored​ in it
5. The “dep” instruction is responsible for finding which graph node to execute next given what
was currently modified